### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by sdf5**

### CODE REVIEW

**Comment by rfreemobile** [comment]
Please fix also the title in the commit itself, the commit still is "Fir...".

git -S --amend    and push again. Easy since it's just 1 commit.

**Review by rfreemobile — CHANGES_REQUESTED** [review]
just fix the title of commit with git -S --amend,
besides that seems ok

File: src/test/utils_check.cpp [hunk 1]
```diff
@@ -427,7 +427,7 @@ TEST(utils_check, reasonable_size_containers_test) {
 		std::multimap<char, std::array<long double, 100u> > too_big_map_by_size;
 		for(size_t i=0; i<size; i++)
 		{
-			too_big_map_by_size.insert(std::make_pair('a', std::array<long double, 100u>{}));
+			too_big_map_by_size.insert(std::make_pair('a', std::array<long double, 100u>{0}));
```
Comment (neodr) [inline src/test/utils_check.cpp]: in clang-4 is warning with this {0}

File: src/tnetdbg.hpp [hunk 2]
```diff
@@ -173,6 +173,23 @@ auto constexpr debug_config_erro_backtrace_level = 128; ///< the backtrace level
...
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wterminate"
+#endif
+	throw ex;
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic pop
+#endif
+}
+
+#define _throw_error_or_abort( EXCEPT ) do { auto except_var = (EXCEPT);  \
```
Comment (rfree-d) [inline src/tnetdbg.hpp]: /***
 * This function tries to throw, BUT ALSO it is O.K. with possible ABORT, due to noexcept on the function we are in
 * it exists to e.g. silence some compilers warning about possible throw inside noexcept - we know, and we are o.k. with it turning into abort
 * use it in macros that should try to throw but abort if throwing is not possible
*/

File: src/utils/check.hpp [hunk 3]
```diff
@@ -152,16 +152,16 @@ class err_check_extern_soft final : public err_check_extern, public err_check_so
 // -------------------------------------------------------------------
 
 /// err_check_prog?
```
Comment (rfree-d) [inline src/utils/check.hpp]: // This macros can throw - BUT ALSO CAN ABORT - if used in noexcept function

File: src/utils/privileges.cpp [hunk 4]
```diff
@@ -55,7 +55,7 @@ std::string get_security_info(bool verbose) noexcept {
 		return oss.str();
 	} catch(...) {
 		_erro("Unexpected error");
-		throw; // trying to rethrow, will cause abort probably (if noexcept)
+		_check_abort(0); // trying to rethrow, will cause abort probably (if noexcept)
```
Comment (rfree-d) [inline src/utils/privileges.cpp]: (false)

File: src/utils/check.hpp [hunk 5]
```diff
@@ -220,7 +220,7 @@ template<typename TC> class c_ig final {
 					_erro("Pre-condition check caused error, for precondition from: "
 						<< m_fromfile << ":" << m_fromline
 					);
-					throw;
+					_check_abort(0);
```
Comment (rfree-d) [inline src/utils/check.hpp]: (false)

File: src/utils/check.hpp [hunk 6]
```diff
@@ -233,7 +233,7 @@ template<typename TC> class c_ig final {
 					_erro("Post-condition check caused error, for precondition from: "
 						<< m_fromfile << ":" << m_fromline
 					);
-					throw;
+					_check_abort(0);
```
Comment (rfree-d) [inline src/utils/check.hpp]: (false)

File: src/test/utils_check.cpp [hunk 7]
```diff
@@ -427,7 +427,7 @@ TEST(utils_check, reasonable_size_containers_test) {
 		std::multimap<char, std::array<long double, 100u> > too_big_map_by_size;
 		for(size_t i=0; i<size; i++)
 		{
-			too_big_map_by_size.insert(std::make_pair('a', std::array<long double, 100u>{0}));
+		        too_big_map_by_size.insert(std::make_pair('a', std::array<long double, 100u>{{0}}));
 		}
```
Comment (rfree-d) [inline src/test/utils_check.cpp]: ^- this ident is ok (tabs versus spaces)?

we use tabs \t

@sdf5

**Review by neodr — APPROVED** [review]

**Pull request merged**